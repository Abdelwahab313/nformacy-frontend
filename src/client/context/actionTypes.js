export const UPDATE_CLIENTS = 'UPDATE_CLIENTS';
export const SET_ERROR_MESSAGE = 'SET_ERROR_MESSAGE';
export const UPDATE_CURRENT_CLIENT = 'UPDATE_CURRENT_CLIENT';
export const VERIFY_CLIENT = 'VERIFY_CLIENT';
export const VERIFY_CURRENT_CLIENT = 'VERIFY_CURRENT_CLIENT';
