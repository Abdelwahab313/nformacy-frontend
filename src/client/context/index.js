import React, { createContext, useContext, useReducer } from 'react';
import {
  SET_ERROR_MESSAGE,
  UPDATE_CLIENTS,
  UPDATE_CURRENT_CLIENT,
  VERIFY_CLIENT,
  VERIFY_CURRENT_CLIENT,
} from './actionTypes';

const ClientContext = createContext();
const defaultValues = {
  clients: [],
  errorMessage: '',
};

const ClientProvider = ({ children, initialValue }) => {
  const { Provider } = ClientContext;
  initialValue = { ...defaultValues, ...initialValue };
  return (
    <Provider value={useReducer(clientReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useClientState = () => {
  const context = useContext(ClientContext);
  if (context === undefined) {
    throw new Error('useClientState must be used within a ClientProvider');
  }
  return context;
};

const clientReducer = (state, action) => {
  switch (action.type) {
    case UPDATE_CLIENTS:
      return {
        ...state,
        clients: action.payload,
      };
    case UPDATE_CURRENT_CLIENT:
      return {
        ...state,
        currentClient: action.payload,
      };
    case VERIFY_CURRENT_CLIENT:
      const currentClient = state.currentClient;
      currentClient.verified = true;
      return {
        ...state,
        currentClient: currentClient,
      };
    case VERIFY_CLIENT:
      const newClients = state.clients.map((client) => {
        if (client[0] === action.payload.uuid) {
          client[6] = true;
          return client;
        } else {
          return client;
        }
      });
      return {
        ...state,
        clients: newClients,
      };
    case SET_ERROR_MESSAGE:
      return {
        ...state,
        errorMessage: action.payload,
      };
  }
};

export { ClientProvider, useClientState };
