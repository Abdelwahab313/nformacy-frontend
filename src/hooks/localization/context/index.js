import React, { createContext, useContext, useReducer } from 'react';

const LocaleContext = createContext();
const DEFAULT_LOCALE = 'en';
const LOCALES = {
  en: 'en',
  ar: 'ar',
};
export const CHANGE_LOCALE = 'CHANGE_LOCALE';

export const useLocaleContext = () => {
  const context = useContext(LocaleContext);
  if (context === undefined)
    throw new Error('useLocale should only be used within locale provider');
  return context;
};

export const LocaleProvider = ({
  children,
  initialLocale = DEFAULT_LOCALE,
}) => {
  const { Provider } = LocaleContext;
  return (
    <Provider value={useReducer(localeReducer, initialLocale)}>{children}</Provider>
  );
};

const localeReducer = (state, action) => {
  if (action.type === CHANGE_LOCALE) {
    return action.payload.locale;
  }
  return LOCALES.en;
};
