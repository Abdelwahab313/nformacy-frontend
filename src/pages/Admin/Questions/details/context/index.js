import React, { createContext, useContext, useReducer } from 'react';

const defaultValues = {
  questionDetails: '',
};

const QuestionContext = createContext();

const QuestionProvider = ({ children, initialValue = {} }) => {
  initialValue = { ...defaultValues, ...initialValue };
  const { Provider } = QuestionContext;
  return (
    <Provider value={useReducer(QuestionReducer, initialValue)}>
      {children}
    </Provider>
  );
};

const useQuestionContext = () => {
  const context = useContext(QuestionContext);
  if (context === undefined) {
    throw new Error('useQuestionContext must be used within a AuthProvider');
  }
  return context;
};

export const QuestionActionTypes = {
  UPDATE_QUESTION: 'UPDATE_QUESTION',
};

const QuestionReducer = (state, action) => {
  switch (action.type) {
    case QuestionActionTypes.UPDATE_QUESTION:
      return { ...state, questionDetails: action.payload };

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export { useQuestionContext as useAuth, QuestionProvider as AuthProvider };
