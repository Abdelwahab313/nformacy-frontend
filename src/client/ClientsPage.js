import React, { useRef, useState } from 'react';
import '../styles/client.css';
import useClientsFetcher from './hooks/ClientsFetcher';
import Grid from '@material-ui/core/Grid';
import { MapWithMultipleMarkers } from '../components/GoogleMap';
import { GOOGLE_MAPS_API_KEY, table_localization } from '../settings';
import CircularProgress from '@material-ui/core/CircularProgress';
import ClientStatus from './components/ClientStatus';
import { Redirect } from 'react-router';
import ErrorDialog from '../components/errors/ErrorDialog';
import { useClientState } from './context';
import {
  SET_ERROR_MESSAGE,
  SET_UPDATE_DIALOG_OPEN,
} from '../salesRep/context/contextAction';
import useClientsPageStyle from './styles/clientsPage';
import MUIDataTable from 'mui-datatables';
import { tableLabels } from '../constants/tableLocalization';
import TableSelectedToolBar from './components/TableSelectedToolBar';
import { VERIFY_CLIENT } from './context/actionTypes';

const ClientsPage = () => {
  const [{ clients, errorMessage }, dispatch] = useClientState();
  const [redirect, setRedirect] = useState(false);
  const [redirectTo, setRedirectTo] = useState();
  let selectedClient = useRef(undefined);
  const {
    clientsLoading,
    locations,
    handleOnStateChanged,
  } = useClientsFetcher();

  const classes = useClientsPageStyle();

  if (redirect) {
    return <Redirect push to={redirectTo} />;
  }
  if (clientsLoading) {
    return (
      <div className={classes.progressContainer}>
        <CircularProgress />
      </div>
    );
  } else {
    return (
      <div dir='rtl' className={classes.root}>
        {errorMessage && (
          <ErrorDialog
            message={errorMessage}
            close={() => {
              dispatch({
                type: SET_ERROR_MESSAGE,
                payload: '',
              });
            }}
          />
        )}
        <Grid
          className={`tableContainer ${
            clients.length === 0
              ? classes.tableContainerFW
              : classes.tableContainer
          }`}>
          <MUIDataTable
            title={'العملاء'}
            data={clients}
            columns={[
              {
                name: 'id',
                options: {
                  filter: false,
                  display: 'excluded',
                },
              },
              {
                name: 'أسم المكان',
                options: {
                  filter: false,
                },
              },
              {
                name: 'أسم المدير',
                options: {
                  filter: false,
                },
              },
              {
                name: 'العنوان',
                options: {
                  filter: false,
                },
              },
              {
                name: 'صوره العميل',
                options: {
                  display: 'excluded',
                  filter: false,
                },
              },
              {
                name: 'رقم الهاتف',
                options: {
                  filter: false,
                },
              },
              {
                name: 'الحاله',
                options: {
                  filterOptions: {
                    names: ['مُوثق', 'غير مُوثق'],
                    logic(status, filters) {
                      if (filters[0] === 'مُوثق') {
                        return status !== true;
                      } else if (filters[0] === 'غير مُوثق') {
                        return status !== false;
                      }
                      return false;
                    },
                  },
                  filter: true,
                  customBodyRender: (value, tableMeta, updateValue) => {
                    return (
                      <ClientStatus
                        status={tableMeta.rowData[6]}
                        clientName={tableMeta.rowData[1]}
                        uuid={tableMeta.rowData[0]}
                        onStateChange={() =>
                          dispatch({
                            type: VERIFY_CLIENT,
                            payload: { uuid: tableMeta.rowData[0] },
                          })
                        }
                      />
                    );
                  },
                },
              },
              {
                name: 'تاريخ الاضافه',
                options: {
                  filter: false,
                  customBodyRender: (value, tableMeta, updateValue) => {
                    const date = new Date(value);
                    return (
                      <p>{`${date.toLocaleString('en-GB', {
                        hour12: true,
                      })}`}</p>
                    );
                  },
                },
              },
            ]}
            options={{
              selectableRowsOnClick: true,
              selectableRows: 'single',
              responsive: 'scrollMaxHeight',
              filter: true,
              rowsPerPageOptions: [5, 10, 20],
              customToolbarSelect: (
                selectedRows,
                displayData,
                setSelectedRows,
              ) => {
                const selectedClientIndex = selectedRows.data[0].dataIndex;
                selectedClient.current = clients[selectedClientIndex];
                return (
                  <TableSelectedToolBar
                    onView={() => {
                      setRedirectTo(`${selectedClient.current[0]}`);
                      setRedirect(true);
                    }}
                    onUpdate={() =>
                      dispatch({ type: SET_UPDATE_DIALOG_OPEN, payload: true })
                    }
                  />
                );
              },
              ...tableLabels,
            }}
          />
        </Grid>
        {clients.length !== 0 && (
          <Grid
            id={'mapContainer'}
            item
            lg={4}
            className={classes.mapContainer}>
            <MapWithMultipleMarkers
              className={classes.map}
              markers={locations}
              isMarkerShown
              googleMapURL={`https://maps.googleapis.com/maps/api/js?v=3.exp&key=${GOOGLE_MAPS_API_KEY}`}
              loadingElement={<div style={{ height: '100%' }} />}
              containerElement={
                <div
                  style={{ display: 'inline-block clear', height: '100%' }}
                />
              }
              mapElement={<div className={'mapElement'} />}
            />
          </Grid>
        )}
      </div>
    );
  }
};

export default ClientsPage;
