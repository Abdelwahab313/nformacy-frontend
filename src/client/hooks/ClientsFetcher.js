import { useEffect, useState } from 'react';
import { useAuth } from '../../auth/auth';
import { fetchClients } from '../clientsApi';
import { default_location } from '../../settings';
import { cloneDeep } from 'lodash';
import { useClientState } from '../context';
import { UPDATE_CLIENTS } from '../context/actionTypes';
import { SET_ERROR_MESSAGE } from '../context/actionTypes';
import { convertObjectToArray } from '../utils';

const ClientsFetcher = () => {
  const [{ clients }, dispatch] = useClientState();
  const [clientsLoading, setClientsLoading] = useState(false);
  const [locations, setLocations] = useState([]);
  const { authTokens, setAuthTokens } = useAuth();

  function handleOnStateChanged(uuid) {
    const updatedClientIndex = clients.findIndex(
      (client) => client.uuid === uuid,
    );
    const tempClientState = cloneDeep(clients);
    tempClientState[updatedClientIndex].verified = true;
    dispatch({ type: UPDATE_CLIENTS, payload: tempClientState });
  }

  function getFirstPhoneNumbers(fetchedClients) {
    for (let i = 0; i < fetchedClients.length; i++) {
      fetchedClients[i].contacts = fetchedClients[i].contacts[0].phone_number;
    }
  }

  const adaptMapsLocation = (lat, long, name) => {
    return {
      lat: parseFloat(lat),
      lng: parseFloat(long),
      label: name,
    };
  };

  function constructLocationsList(fetchedClients) {
    const extractedLocations = [];
    for (let i = 0; i < fetchedClients.length; i++) {
      extractedLocations.push(
        adaptMapsLocation(
          fetchedClients[i].location.coordinates[0],
          fetchedClients[i].location.coordinates[1],
          fetchedClients[i].name,
        ),
      );
    }
    if (fetchedClients.length === 0) {
      extractedLocations.push(default_location);
    }
    setLocations(extractedLocations);
  }

  useEffect(() => {
    setClientsLoading(true);
    fetchClients(authTokens)
      .then((res) => {
        const fetchedClients = res.data;
        fetchedClients.sort(
          (a, b) => new Date(a.created) - new Date(b.created),
        );
        getFirstPhoneNumbers(fetchedClients);
        constructLocationsList(fetchedClients);

        const fetchedClientsAsArray = fetchedClients.map((client) =>
          convertObjectToArray(client),
        );
        dispatch({ type: UPDATE_CLIENTS, payload: fetchedClientsAsArray });
      })
      .catch((reason) => {
        if (reason.message === 'Network Error') {
          dispatch({
            type: SET_ERROR_MESSAGE,
            payload: 'حدث خطأ أثناء الاتصال بالخادم',
          });
        } else if (reason.response.status === 401) {
          localStorage.removeItem('tokens');
          localStorage.removeItem('users');
          setAuthTokens();
        }
      })
      .finally(() => {
        setClientsLoading(false);
      });
  }, []);

  return {
    clientsLoading,
    locations,
    handleOnStateChanged,
  };
};

export default ClientsFetcher;
