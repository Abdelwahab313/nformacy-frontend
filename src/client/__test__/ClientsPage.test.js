import { render } from '@testing-library/react';
import * as AuthContextModule from '../../auth/auth';
import { AuthContext } from '../../auth/auth';
import React from 'react';
import { ClientProvider } from '../context';
import ClientsPage from '../ClientsPage';
import ClientsFetcher from '../hooks/ClientsFetcher';

jest.mock('../hooks/ClientsFetcher');
describe('ClientsFetcher Presentation', () => {
  it('should render clients page correctly', () => {
    jest.spyOn(AuthContextModule, 'useAuth').mockImplementation(() => ({
      authTokens: { access_token: 'testToken' },
    }));
    ClientsFetcher.mockImplementationOnce(() => [false, 'setter']);
    const clients = [
      [
        'test_uuid',
        'test_name',
        'test_owner_name',
        'test_address',
        'image_link',
        true,
        { coordinates: [200, 200] },
        [{ phone_number: '0123456789' }],
      ],
    ];
    const { asFragment } = render(
      <AuthContext.Provider>
        <ClientProvider initialValue={{ clients: clients }}>
          <ClientsPage />
        </ClientProvider>
      </AuthContext.Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
