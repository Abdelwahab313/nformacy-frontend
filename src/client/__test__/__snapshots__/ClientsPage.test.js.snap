// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientsFetcher Presentation should render clients page correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
    dir="rtl"
  >
    <div
      class="MuiGrid-root tableContainer makeStyles-tableContainer-6"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation4 MUIDataTable-paper-113 MuiPaper-rounded"
      >
        <div
          aria-label="Table Toolbar"
          class="MuiToolbar-root MuiToolbar-regular MUIDataTableToolbar-root-148 MuiToolbar-gutters"
          role="toolbar"
        >
          <div
            class="MUIDataTableToolbar-left-149"
          >
            <div
              aria-hidden="true"
              class="MUIDataTableToolbar-titleRoot-151"
            >
              <h6
                class="MuiTypography-root MUIDataTableToolbar-titleText-152 MuiTypography-h6"
              >
                العملاء
              </h6>
            </div>
          </div>
          <div
            class="MUIDataTableToolbar-actions-150"
          >
            <button
              aria-label="بحث"
              class="MuiButtonBase-root MuiIconButton-root MUIDataTableToolbar-icon-153"
              data-testid="بحث-iconButton"
              tabindex="0"
              title="بحث"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="CSV تحميل"
              class="MuiButtonBase-root MuiIconButton-root MUIDataTableToolbar-icon-153"
              data-testid="CSV تحميل-iconButton"
              tabindex="0"
              title="CSV تحميل"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM17 13l-5 5-5-5h3V9h4v4h3z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <span>
              <span>
                <button
                  aria-label="طباعه"
                  class="MuiButtonBase-root MuiIconButton-root MUIDataTableToolbar-icon-153"
                  data-testid="طباعه-iconButton"
                  tabindex="0"
                  title="طباعه"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 8H5c-1.66 0-3 1.34-3 3v6h4v4h12v-4h4v-6c0-1.66-1.34-3-3-3zm-3 11H8v-5h8v5zm3-7c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm-1-9H6v4h12V3z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </span>
            </span>
            <span>
              <button
                aria-label="عرض"
                class="MuiButtonBase-root MuiIconButton-root MUIDataTableToolbar-icon-153"
                data-testid="عرض-iconButton"
                tabindex="0"
                title="عرض"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 18h5V5h-5v13zm-6 0h5V5H4v13zM16 5v13h5V5h-5z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </span>
            <span>
              <button
                aria-label="تصنيف"
                class="MuiButtonBase-root MuiIconButton-root MUIDataTableToolbar-icon-153"
                data-testid="تصنيف-iconButton"
                tabindex="0"
                title="تصنيف"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </span>
          </div>
        </div>
        <div
          class="MUIDataTableFilterList-root-223"
        />
        <div
          class="MUIDataTable-responsiveScrollMaxHeight-116"
          style="position: relative;"
        >
          <table
            class="MuiTable-root MUIDataTable-tableRoot-114"
            role="grid"
            tabindex="0"
          >
            <caption
              class="MUIDataTable-caption-119"
            >
              العملاء
            </caption>
            <thead
              class="MuiTableHead-root MUIDataTableHead-main-227"
            >
              <tr
                class="MuiTableRow-root MUIDataTableHeadRow-root-230 MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MUIDataTableSelectCell-root-236 MUIDataTableSelectCell-fixedHeaderCommon-238 MUIDataTableSelectCell-fixedHeaderXAxis-239 MUIDataTableSelectCell-fixedHeaderYAxis-240 MUIDataTableSelectCell-headerCell-244 MuiTableCell-paddingCheckbox"
                  scope="col"
                >
                  <div
                    style="display: flex; align-items: center;"
                  />
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-261 MUIDataTableHeadCell-fixedHeaderCommon-263 MUIDataTableHeadCell-fixedHeaderXAxis-264 MUIDataTableHeadCell-fixedHeaderYAxis-265"
                  scope="col"
                >
                  <span
                    class="MUIDataTableHeadCell-toolButton-272"
                    role="button"
                    tabindex="0"
                    title="ترتيب حسب أسم المكان"
                  >
                    <div
                      class="MUIDataTableHeadCell-data-268"
                    >
                      أسم المكان
                    </div>
                    <div
                      class="MUIDataTableHeadCell-sortAction-269"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-270"
                        role="button"
                        tabindex="0"
                      />
                    </div>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-261 MUIDataTableHeadCell-fixedHeaderCommon-263 MUIDataTableHeadCell-fixedHeaderXAxis-264 MUIDataTableHeadCell-fixedHeaderYAxis-265"
                  scope="col"
                >
                  <span
                    class="MUIDataTableHeadCell-toolButton-272"
                    role="button"
                    tabindex="0"
                    title="ترتيب حسب أسم المدير"
                  >
                    <div
                      class="MUIDataTableHeadCell-data-268"
                    >
                      أسم المدير
                    </div>
                    <div
                      class="MUIDataTableHeadCell-sortAction-269"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-270"
                        role="button"
                        tabindex="0"
                      />
                    </div>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-261 MUIDataTableHeadCell-fixedHeaderCommon-263 MUIDataTableHeadCell-fixedHeaderXAxis-264 MUIDataTableHeadCell-fixedHeaderYAxis-265"
                  scope="col"
                >
                  <span
                    class="MUIDataTableHeadCell-toolButton-272"
                    role="button"
                    tabindex="0"
                    title="ترتيب حسب العنوان"
                  >
                    <div
                      class="MUIDataTableHeadCell-data-268"
                    >
                      العنوان
                    </div>
                    <div
                      class="MUIDataTableHeadCell-sortAction-269"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-270"
                        role="button"
                        tabindex="0"
                      />
                    </div>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-261 MUIDataTableHeadCell-fixedHeaderCommon-263 MUIDataTableHeadCell-fixedHeaderXAxis-264 MUIDataTableHeadCell-fixedHeaderYAxis-265"
                  scope="col"
                >
                  <span
                    class="MUIDataTableHeadCell-toolButton-272"
                    role="button"
                    tabindex="0"
                    title="ترتيب حسب رقم الهاتف"
                  >
                    <div
                      class="MUIDataTableHeadCell-data-268"
                    >
                      رقم الهاتف
                    </div>
                    <div
                      class="MUIDataTableHeadCell-sortAction-269"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-270"
                        role="button"
                        tabindex="0"
                      />
                    </div>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-261 MUIDataTableHeadCell-fixedHeaderCommon-263 MUIDataTableHeadCell-fixedHeaderXAxis-264 MUIDataTableHeadCell-fixedHeaderYAxis-265"
                  scope="col"
                >
                  <span
                    class="MUIDataTableHeadCell-toolButton-272"
                    role="button"
                    tabindex="0"
                    title="ترتيب حسب الحاله"
                  >
                    <div
                      class="MUIDataTableHeadCell-data-268"
                    >
                      الحاله
                    </div>
                    <div
                      class="MUIDataTableHeadCell-sortAction-269"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-270"
                        role="button"
                        tabindex="0"
                      />
                    </div>
                  </span>
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MUIDataTableHeadCell-root-261 MUIDataTableHeadCell-fixedHeaderCommon-263 MUIDataTableHeadCell-fixedHeaderXAxis-264 MUIDataTableHeadCell-fixedHeaderYAxis-265"
                  scope="col"
                >
                  <span
                    class="MUIDataTableHeadCell-toolButton-272"
                    role="button"
                    tabindex="0"
                    title="ترتيب حسب تاريخ الاضافه"
                  >
                    <div
                      class="MUIDataTableHeadCell-data-268"
                    >
                      تاريخ الاضافه
                    </div>
                    <div
                      class="MUIDataTableHeadCell-sortAction-269"
                    >
                      <span
                        aria-disabled="false"
                        class="MuiButtonBase-root MuiTableSortLabel-root MUIDataTableHeadCell-sortLabelRoot-270"
                        role="button"
                        tabindex="0"
                      />
                    </div>
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root MUIDataTableBodyRow-root-283 MUIDataTableBodyRow-hover-284 MUIDataTableBodyRow-hoverCursor-285 MuiTableRow-hover"
                data-testid="MUIDataTableBodyRow-0"
                id="MUIDataTableBodyRow-0"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableSelectCell-root-236 MUIDataTableSelectCell-fixedHeaderCommon-238 MUIDataTableSelectCell-fixedHeaderXAxis-239 MUIDataTableSelectCell-fixedHeaderYAxis-240 MuiTableCell-paddingCheckbox"
                >
                  <div
                    style="display: flex; align-items: center;"
                  >
                    <span
                      aria-disabled="false"
                      class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-293 MuiCheckbox-root MUIDataTableSelectCell-checkboxRoot-246 MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <input
                          class="PrivateSwitchBase-input-296"
                          data-indeterminate="false"
                          id="MUIDataTableSelectCell-0"
                          type="checkbox"
                          value=""
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </span>
                  </div>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-297 MUIDataTableBodyCell-cellHide-298"
                >
                  أسم المكان
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-297"
                  data-testid="MuiDataTableBodyCell-1-0"
                >
                  test_name
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-297 MUIDataTableBodyCell-cellHide-298"
                >
                  أسم المدير
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-297"
                  data-testid="MuiDataTableBodyCell-2-0"
                >
                  test_owner_name
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-297 MUIDataTableBodyCell-cellHide-298"
                >
                  العنوان
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-297"
                  data-testid="MuiDataTableBodyCell-3-0"
                >
                  test_address
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-297 MUIDataTableBodyCell-cellHide-298"
                >
                  رقم الهاتف
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-297"
                  data-testid="MuiDataTableBodyCell-5-0"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-297 MUIDataTableBodyCell-cellHide-298"
                >
                  الحاله
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-297"
                  data-testid="MuiDataTableBodyCell-6-0"
                >
                  <p
                    class="makeStyles-verified-301"
                  >
                    مُوثق
                  </p>
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-297 MUIDataTableBodyCell-cellHide-298"
                >
                  تاريخ الاضافه
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MUIDataTableBodyCell-root-297"
                  data-testid="MuiDataTableBodyCell-7-0"
                >
                  <p>
                    Invalid Date
                  </p>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <table
          class="MuiTable-root"
        >
          <tfoot
            class="MuiTableFooter-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-footer"
            >
              <td
                class="MuiTableCell-root MuiTableCell-footer MuiTablePagination-root MUIDataTablePagination-root-304"
                colspan="1000"
              >
                <div
                  class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MUIDataTablePagination-toolbar-305 MuiToolbar-gutters"
                >
                  <div
                    class="MuiTablePagination-spacer"
                  />
                  <p
                    class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
                  >
                    عدد الصفوف:
                  </p>
                  <div
                    class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot MUIDataTablePagination-selectRoot-306"
                  >
                    <div
                      aria-haspopup="listbox"
                      aria-labelledby=" pagination-rows"
                      class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input MuiInputBase-inputSelect"
                      data-testid="pagination-rows"
                      id="pagination-rows"
                      role="button"
                      tabindex="0"
                    >
                      10
                    </div>
                    <input
                      type="hidden"
                      value="10"
                    />
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                      focusable="false"
                      role="presentation"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </div>
                  <p
                    class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
                  >
                    1-1 من 1
                  </p>
                  <div
                    class="MuiTablePagination-actions"
                  >
                    <button
                      aria-label="الصفحه السابقه"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                      data-testid="pagination-back"
                      disabled=""
                      id="pagination-back"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                          />
                        </svg>
                      </span>
                    </button>
                    <button
                      aria-label="الصفحه التاليه"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
                      data-testid="pagination-next"
                      disabled=""
                      id="pagination-next"
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                </div>
              </td>
            </tr>
          </tfoot>
        </table>
        <div
          aria-live="polite"
          class="MUIDataTable-liveAnnounce-120"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-mapContainer-8 MuiGrid-item MuiGrid-grid-lg-4"
      id="mapContainer"
    >
      <div
        style="height: 100%;"
      />
    </div>
  </div>
</DocumentFragment>
`;
